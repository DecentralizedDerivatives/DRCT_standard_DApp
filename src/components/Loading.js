import React from 'react';

export const Loading = () => {
  return (
    <div>
      <img className='loading-spinner' alt='Loading' src={require("../imgs/spinner.gif")} />
      <h5>Loading ... </h5>
    </div>
  );
};

export default Loading;
