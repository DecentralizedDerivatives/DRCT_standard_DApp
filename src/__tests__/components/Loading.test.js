import Loading from '../../components/Loading';

describe('<Loading />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <Loading store={initStore()} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
