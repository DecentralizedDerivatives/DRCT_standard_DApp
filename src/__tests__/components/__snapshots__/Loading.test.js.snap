// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Loading"
          className="loading-spinner"
          src="spinner.gif"
        />,
        <h5>
          Loading ... 
        </h5>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Loading",
          "className": "loading-spinner",
          "src": "spinner.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading ... ",
        },
        "ref": null,
        "rendered": "Loading ... ",
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Loading"
            className="loading-spinner"
            src="spinner.gif"
          />,
          <h5>
            Loading ... 
          </h5>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Loading",
            "className": "loading-spinner",
            "src": "spinner.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading ... ",
          },
          "ref": null,
          "rendered": "Loading ... ",
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
